src/Framework/PaymentGateways/CommandHandlers/PaymentHandler.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Give\Framework\PaymentGateways\CommandHandlers;
+
+use Give\Framework\PaymentGateways\Commands\PaymentCommand;
+
+abstract class PaymentHandler
+{
+    /**
+     * @unreleased
+     * @return string
+     */
+    abstract protected function getPaymentStatus();
+
+    /**
+     * @unreleased
+     *
+     * @param  PaymentCommand  $paymentCommand
+     * @param  int  $donationId
+     * @return void
+     */
+    public function __invoke(PaymentCommand $paymentCommand, $donationId)
```

Comment:
One thing that makes this abstraction different, is we are generalizing the `$paymentCommand` in our handler instead of being specific about the command the handler consumes.  I realize that might be a trade-off for the benefits of this abstraction though so open for discussion.

Reply:
What if me moved the `$paymentCommand` to an overloaded constructor?

Reply:
Yeah that will work.  if we're going to enforce a constructor then we should probably turn `__invoke` into `handle`

Reply:
That makes sense.

Reply:
Resolved by bee2ae01a.